spec/controllers/students_controller_spec.rb

```diff
@@ -604,6 +604,41 @@ def make_request(options = { student_id: nil, format: :pdf, from_date: '08/15/20
           expect(assigns(:event_notes)).to include(note)
           expect(assigns(:event_notes)).not_to include(restricted_note)
         end
+
+        it 'assigns the student\'s school years correctly' do
+          incident = FactoryGirl.create(:discipline_incident, student: student, occurred_at: '2015-08-15')
+          absence = FactoryGirl.create(:absence, student: student, occurred_at: '2015-08-16')
+          tardy = FactoryGirl.create(:tardy, student: student, occurred_at: '2015-08-17')
+          make_request({ student_id: student.id, format: :pdf, from_date: '08/15/2015', to_date: '03/16/2017' })
+
+          expect(assigns(:student_school_years)[0].discipline_incidents).to include(incident)
+          expect(assigns(:student_school_years)[0].absences).to include(absence)
+          expect(assigns(:student_school_years)[0].tardies).to include(tardy)
+        end
+
+        it 'assigns the student\'s discipline incidents correctly' do
+          incident = FactoryGirl.create(:discipline_incident, student: student, occurred_at: '2015-08-15')
+          old_incident = FactoryGirl.create(:discipline_incident, student: student, occurred_at: '2015-08-14')
+          make_request({ student_id: student.id, format: :pdf, from_date: '08/15/2015', to_date: '03/16/2017' })
+
+          expect(assigns(:discipline_incidents)).to include(incident)
+          expect(assigns(:discipline_incidents)).not_to include(old_incident)
+        end
+
+        it 'assigns the student\'s assesments correctly' do
+          assessment = FactoryGirl.create(:assessment, :access)
+          student_assessment = FactoryGirl.create(:access, student: student, assessment: assessment, date_taken: '2016-08-16')
+          assessment = FactoryGirl.create(:assessment, :math)
+          student_assessment = FactoryGirl.create(:star_math_assessment, student: student, assessment: assessment, date_taken: '2017-02-16')
+          make_request({ student_id: student.id, format: :pdf, from_date: '08/15/2015', to_date: '03/16/2017' })
+
+          expect(assigns(:student_assessments)).to include("ACCESS Composite")
+          expect(assigns(:student_assessments)["ACCESS Composite"]).to be_kind_of(Array)
+          expect(assigns(:student_assessments)["ACCESS Composite"]).to eq([["2016-08-16 00:00:00 UTC", nil]])
+          expect(assigns(:student_assessments)).to include(" Mathematics")
```

Comment:
@gabivoicu Why is there a leading space in `" Mathematics"`? That seems like an easy thing for a developer to miss . . . is the code set up to expect a leading space somehow or is that an artifact of the spec code?

Reply:
@alexsoble The leading space was there because there was no family name for that test type. Looking at the factories again, I realized I should have used
`assessment = FactoryGirl.create(:assessment, :star)`
so now those tests will have a family, and it will be `STAR`.

Thanks for pointing that out!